src/lib/flight_tasks/tasks/Orbit/FlightTaskOrbit.cpp

```diff
@@ -172,12 +180,14 @@ bool FlightTaskOrbit::update()
 	setVelocity(v);
 
 	Vector2f center_to_position = Vector2f(_position) - _center;
+	Vector2f start_to_circle = (_r - center_to_position.norm()) * center_to_position.unit_or_zero();
 
-	// make vehicle front always point towards the center
-	_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) + M_PI_F;
+	if (start_to_circle.norm() < 0.5f) {
```

Comment:
This is overriding the logic of the motion primitive `StraightLine _circle_approach_line`. I can only guess that according to your taste the approach takes too much time to slow down at the end. But then it's the fault of the `StraightLine` implementation which we want to fill with the jerk optimized trajectory generation anyways very soon and we should not interfere with its operation locally here.

Reply:
I added that not because the slow down took too much time, but because switching from one orbit command to another orbit command made the drone brake down (same orbit center, same radius, but different yaw behaviour). But thinking about it now, this braking down seems reasonable.